hooks/useTokenBalance.tsx

```diff
@@ -0,0 +1,45 @@
+import { TokenListItem } from "@/hooks/useTokenList";
+import { BigNumber } from "@ethersproject/bignumber";
+import { readContracts } from "@wagmi/core";
+import useSWR from "swr";
+import { useAccount, useNetwork } from "wagmi";
+
+export const ERC20_ABI = [
+  "function balanceOf(address owner) view returns (uint256 balance)",
+];
+
+const useTokenList = ({ tokenList }: { tokenList: TokenListItem[] }) => {
+  const { chain } = useNetwork();
+  const { address } = useAccount();
+
+  const chainId = chain?.id || 1; // default to mainnet if no chain id
+  const { isLoading, error, data } = useSWR(
+    `userTokenList-${chainId}-${tokenList.length}`,
+    async () => {
+      if (!tokenList || tokenList.length === 0) {
+        return;
+      }
+      const readContractsArgs = tokenList.map((token) => {
+        return {
+          abi: ERC20_ABI,
+          functionName: "balanceOf",
+          address: token.address,
+          args: [address],
+        };
+      });
+      const data = await readContracts({
```

Comment:
is it the case that this will make n rpc calls for n tokens? perhaps we can consider [`watchMulticall`](https://wagmi.sh/core/actions/watchMulticall) instead, any thoughts?

Reply:
This is good as-is, check out the conversation here + the linked wagmi discussion where I asked them https://github.com/ScopeLift/railgun-shielder/pull/32#discussion_r1066225006

Reply:
ah gotcha thanks